rd-kt/rd-framework/src/main/kotlin/com/jetbrains/rd/framework/Interfaces.kt

```diff
@@ -38,6 +38,10 @@ interface IProtocol : IRdDynamic {
     val contexts : ProtocolContexts
 
     val extCreated: ISignal<ExtCreationInfo>
+
+    fun <T: Any> getOrCreateExtension(clazz: KClass<T>, create: () -> T): T
+    fun <T: Any> tryGetExtension(clazz: KClass<T>): T?
+    fun <T: Any> createExtensionOrThrow(clazz: KClass<T>, create: () -> T): T
```

Comment:
Is it still needed?

Reply:
Yes, it is. `getOrCreateExtension` for an obvious reason of implementing all extension functions without casting `IProtocol` to `Protocol` every time. `tryGetExtension` is needed to support `getOrNull` method (even if we remove this method, then this protocol method will be used instead). And `createExtensionOrThrow` is very important for some very specific things like `ProtocolExtListenerManager`.

Reply:
I don't see usages of `createExtensionOrThrow` neither in rd, nor in `ProtocolExtListenerManager`

Reply:
It is not used in rd indeed, but it is used in `ProtocolExtListenerManager` here: https://jetbrains.team/p/ij/reviews/18365/files?file=%2Fplatform%2Frd-platform%2Fsrc%2Fcom%2Fjetbrains%2Frd%2Fprotocol%2FProtocolExtListenerManager.kt&new-line=181

Reply:
Fixed